test/test_displacement.py

```diff
@@ -0,0 +1,96 @@
+#  Copyright (c) 2025 National Land Survey of Finland (Maanmittauslaitos)
+#
+#  This file is part of geogen-algorithms.
+#
+#  This source code is licensed under the MIT license found in the
+#  LICENSE file in the root directory of this source tree.
+
+import geopandas as gpd
+import pytest
+from pandas.testing import assert_frame_equal
+from shapely import equals_exact
+from shapely.geometry import Point
+
+from geogenalg import displacement
+
+
+@pytest.mark.parametrize(
+    ("input_gdf", "threshold", "expected_gdf"),
+    [
+        (
+            gpd.GeoDataFrame(geometry=[Point(0, 0)]),
+            1.0,
+            gpd.GeoDataFrame(geometry=[Point(0, 0)]),
+        ),
+        (
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(2, 0)]),
+            1.0,
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(2, 0)]),
+        ),
+        (
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(0.5, 0)]),
+            1.0,
+            gpd.GeoDataFrame(geometry=[Point(-0.25, 0), Point(0.75, 0)]),
+        ),
+        (
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(1, 0), Point(2, 0)]),
+            2.0,
+            gpd.GeoDataFrame(geometry=[Point(-0.5, 0), Point(1, 0), Point(2.5, 0)]),
+        ),
+        (
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(0.5, 0), Point(5, 5)]),
+            1.0,
+            gpd.GeoDataFrame(geometry=[Point(-0.25, 0), Point(0.75, 0), Point(5, 5)]),
+        ),
+        (
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(0, 0), Point(0, 0)]),
+            1.0,
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(0, 0), Point(0, 0)]),
+        ),
+        (
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(1.0, 0)]),
+            1.0,
+            gpd.GeoDataFrame(geometry=[Point(0, 0), Point(1.0, 0)]),
+        ),
+        (
+            gpd.GeoDataFrame(
+                {"id": [5, 4, 6], "category": ["Z", "X", "Y"]},
+                geometry=[Point(0, 0), Point(0.5, 0), Point(5, 5)],
+            ),
+            1.0,
+            gpd.GeoDataFrame(
+                {"id": [5, 4, 6], "category": ["Z", "X", "Y"]},
+                geometry=[Point(-0.25, 0), Point(0.75, 0), Point(5, 5)],
+            ),
+        ),
+    ],
+    ids=[
+        "single_point",
+        "two_far_apart",
+        "two_close_displaced",
+        "three_close_displaced",
+        "two_close_one_far",
+        "three_points_same_location",
+        "distance_exactly_threshold",
+        "three_points_with_attributes",
```

Comment:
You could move this last test case to be it's own separate test and name it for example "test_displace_points_preserves_attributes_intact" or something like that. And then this first test could be "test_displace_points_moves_points_expectedly" or something like that. The point is to make both tests shorter and easier to understand.